PocketKit/Sources/PocketKit/Collections/CollectionViewModel.swift

```diff
@@ -229,37 +237,59 @@ private extension CollectionViewModel {
         return snapshot
     }
 
-    func buildSnapshot() -> Snapshot {
-        var snapshot = Snapshot()
-
-        guard let collection else { return snapshot }
+    /// Builds a snapshot when collection stories are found in Core Data
+    func buildCollectionSnapshot(_ IDs: [NSManagedObjectID]) -> Snapshot {
+        var collectionSnapshot = Snapshot()
+        let storiesSection = Section.collection(collection)
+        collectionSnapshot.appendSections([.collectionHeader, storiesSection])
+        collectionSnapshot.appendItems([.collectionHeader], toSection: .collectionHeader)
 
-        let section: CollectionViewModel.Section = .collection(collection)
-        snapshot.appendSections([.collectionHeader, section])
-        snapshot.appendItems([.collectionHeader], toSection: .collectionHeader)
+        let entities = IDs.map {
+            source.viewObject(id: $0) as! CollectionStory
+        }
 
-        collection.stories.forEach { story in
-            let storyViewModel = CollectionStoryViewModel(story: story)
-            snapshot.appendItems(
-                [.stories(storyViewModel)],
-                toSection: section
+        let models = entities.map {
+            CollectionStoryModel(
+                title: $0.title,
+                publisher: $0.publisher,
+                imageURL: $0.imageUrl,
+                excerpt: $0.excerpt,
+                timeToRead: $0.item?.timeToRead != nil ? Int(truncating: ($0.item?.timeToRead)!) : nil,
+                isCollection: $0.item?.collection != nil
             )
         }
-
-        return snapshot
+        let cells = models.map {
+            Cell.story(CollectionStoryViewModel(storyModel: $0))
+        }
+        collectionSnapshot.appendItems(cells, toSection: storiesSection)
+        return collectionSnapshot
     }
 }
 
 extension CollectionViewModel {
     enum Section: Hashable {
         case loading
         case collectionHeader
-        case collection(CollectionModel)
+        case collection(Collection)
     }
 
     enum Cell: Hashable {
         case loading
         case collectionHeader
-        case stories(CollectionStoryViewModel)
+        case story(CollectionStoryViewModel)
+    }
+}
+
+extension CollectionViewModel: NSFetchedResultsControllerDelegate {
+    func controller(_ controller: NSFetchedResultsController<NSFetchRequestResult>, didChangeContentWith snapshot: NSDiffableDataSourceSnapshotReference) {
+        let IDs: [NSManagedObjectID] = snapshot.itemIdentifiers.compactMap {
+            $0 as? NSManagedObjectID
+        }
+
+        guard !IDs.isEmpty else {
+            return
+        }
+
+        self.snapshot = buildCollectionSnapshot(IDs)
```

Comment:
hmm so, a couple of considerations here:
1. This might change as we add behaviors like saving an item in a collection
2. Until it does, we are not rebuilding, but just building the snapshot. In fact, it won't have any reloaded or reconfigured items. Again, it might change, but I'd like to address this when it does
3. I'd argue that on home we are updating the snapshot. In fact, the method `buildSnapshot` in there, first builds a new snapshot from scratch, using what's in the fetched result VCs, then updates it based on the `snapshot` returned by the delegate method. I am not totally sure that this would be faster than just creating a new one.
Thoughts @bassrock ?

Reply:
@Gio2018 thats fine i dont hold a strong opinion here and trust you on this.